### DISCUSSION TITLE
Home bases for squadrons

### DISCUSSION THREAD

**Discussion by DanAlbert** [Khopa/dcs_liberation#1550] [Dev Diaries]
**The features described in this dev diary are only available in the development branch. You can try them out as part of the preview build, but as always save games are expected to break *often* in this branch.**

Squadrons were first introduced in DCS Liberation 3.0 and 5.0 will continue to build on them by assigning each squadron to a specific base. Each squadron having a consistent base improves the role play of the game by keeping your pilots from teleporting across the map each turn.

To support this, there are some additional changes that are needed for aircraft allocation. Specifically, rather than being assigned to an airbase or carrier, aircraft will belong to a squadron. This way when a squadron is moved to a new base, they move with their aircraft (more on that later).

Later on aircraft may be immediately transferred between squadrons at the same base, but for now the aircraft must be sold and repurchased to move them between squadrons.

## Initial base assignments

Initial locations for each squadron are defined by the campaign based on the role of each squadron. For example, in The Black Sea, the US begins with the following squadrons:

CV: 2 F/A-18C, 2 F-14B, S-3B tanker
LHA: AV-8B, UH-1
Kobuleti: 2 F-16C, F-15C
Kutaisi: E-3A, KC-135, F-15C, C-17
Senaki-Kholki: A-10C, F-15E, F-16C, UH-60

The starting base for each squadron can be changed during air wing configuration at the start of the campaign.

What if you play the campaign with a faction that doesn't have those aircraft? The campaign specifies its preferred aircraft (or even a specific squadron) for each squadron slot, but if none of the preferred types are available a squadron will automatically be created based on the primary task of the squadron.

Using the A-10C squadron at Senaki-Kholkian example, it is defined as (in order):

Primary task: CAS
Secondary tasks: air-to-ground
Aircraft: A-10C Thunderbolt II (Suite 7), A-10C Thunderbolt II (Suite 3)

If the Suite 7 A-10C (the updated DCS module) isn't available, the legacy A-10C module will be used. If neither A-10C is available, the best (with some wiggle room to maintain variety) CAS aircraft will be chosen. If the list of aircraft and roles to try has been exhausted and the faction has no compatible aircraft available, the squadron will simply not be used.

The "secondary tasks" define the other types of missions that squadron can be automatically assigned to. This field does not restrict the player from assigning the aircraft to other mission types; it only influences the auto-planner.

Campaign designers, read the wiki for information on configuring squadron locations.

## Squadron relocation

The squadron's base at the start of the game is not permanent. It can be moved each turn if desired, but it's expected that moves are infrequent. When a move order is given, any unused aircraft belonging to that squadron will be automatically assigned to a package of ferry flights.

Ferry flights can be canceled as desired; they're only for flavor. Regardless of ferry flights, any surviving aircraft will be moved to the new base at the end of the turn.

If missions are canceled, those aircraft can be assigned to missions and their arrival airport will automatically be set to the squadron's new home.

Missions planned before the move order is given will not have their arrival airports altered, but will still be relocated at the end of the turn nonetheless.

A new loadout "DCS Ferry" can be used to configure the loadout that aircraft should use for ferry missions. These are not necessary, but extra gas is always recommended, and you might as well finally use those travel pods!

## Automated moves

Squadrons will only relocate automatically if their base is captured. When this happens they will follow [the same rules that aircraft did in previous versions](https://github.com/dcs-liberation/dcs_liberation/wiki/Base-Capture#unit-retreat-behavior), but all of a squadron's aircraft must retreat to a single base.

In a future release shorter range aircraft may automatically relocate as the front line moves, but as long as squadrons can still reach targets they will stick to their base.

**Comment by Slowdive1123** [comment]
This is an awesome addition!

I also noticed that some squadron management stuff is there when setting up a campaign. Would it be possible to generate a list of available liveries, say from the faction definition and have them available as a drop down to assign to a squadron? It'd also be nice to be able to setup more than one squadron per aircraft type.
